### PR TITLE
EIP-6206: Simplify JUMPF validation

### PR DESCRIPTION
**Pull request by gumb0**
This replaces two requirements on `stack_height` at JUMPF (inequality + equality) with equivalent combination of JUMPF instruction validation requirement and `stack_height` equality requirement.

### CODE REVIEW

File: EIPS/eip-6206.md [hunk 1]
```diff
@@ -35,8 +35,8 @@ A new instruction, `JUMPF (0xb2)`, is introduced.
 Let the definition of `type[i]` be inherited from [EIP-4750](./eip-4750.md) and define `stack_height` to be the height of the stack at a certain instruction during the instruction flow traversal if the operand stack at the start of the function were equal to `type[i].inputs`.
 
 * The immediate argument of `JUMPF` MUST be less than the total number of code sections.
-* The stack height at `JUMPF` MUST be greater than or equal to `type[code_section_index].inputs`.
```
Comment (gumb0) [inline EIPS/eip-6206.md:38]: Old version: 
1. `stack_height >= type[code_section_index].inputs`
2. `type[current_section_index].outputs == stack_height - type[code_section_index].inputs + type[code_section_index].outputs`

Point 2 gives requirement on exact value of `stack_height`, therefore `stack_height` can be expressed in terms of sections inputs/outputs.

File: EIPS/eip-6206.md [hunk 2]
```diff
@@ -35,8 +35,8 @@ A new instruction, `JUMPF (0xb2)`, is introduced.
...
-* `type[current_section_index].outputs` MUST equal `stack_height - type[code_section_index].inputs + type[code_section_index].outputs`. This means that `code_section_index` can output less stack elements than the original code section called by the top element on the return stack if the `current_section_index` code section leaves the delta `type[current_section_index].outputs - type[code_section_index].outputs` element(s) on the stack.
+* For each `JUMPF` instruction `type[current_section_index].outputs` MUST be greater or equal `type[code_section_index].outputs`.
```
Comment (gumb0) [inline EIPS/eip-6206.md:38]: New version:
1. `type[current_section_index].outputs >= type[code_section_index].outputs`
2. `stack_height == type[code_section_index].inputs + type[current_section_index].outputs - type[code_section_index].outputs`

Comment (chfast) [reply EIPS/eip-6206.md:38]: I wander if this is not more clear:
```
stack_height == type[current_section_index].outputs + (type[code_section_index].inputs  - type[code_section_index].outputs)
```

**Review by chfast — APPROVED** [review]

File: EIPS/eip-6206.md (hunk 2 above)
Comment (gumb0) [reply EIPS/eip-6206.md:38]: Maybe slightly. I changed it to this variant (without parentheses)

**Pull request merged**